variables.tf

```diff
@@ -339,21 +339,21 @@ variable "storage" {
 variable "kms" {
   description = <<EOF
     enabled = "Toggle,if set use either the specified KMS key_id or a Domino-generated one"
-    key_id  = optional(string, null)
+    key_id  = optional(list(string), [])
```

Comment:
why does the key id need a list?

Reply:
Explained it in the comment where I pinged you, but not very clearly. To be able to do tests, I set up the test to provision a KMS key outside of the module. However, because we're using the key with a `count` and the key's value cannot be known to terraform prior to apply it either errors or requires a targeted apply of the before running "for real". By making it a list of strings, we can check the length of the list, getting around this terraform restriction.

While I don't like doing obtuse things just for the sake of the test, it also occurs to me that perhaps this change is positive since someone consuming this terraform could conceivably want to do something similar.

If we don't want to do this, I can also just switch the test to do a targeted apply or configure the external kms key with separate terraform (or if you know a better way around this, we can do that).